### PR TITLE
@86epmb28q Added Detection Access Token and Added customizable style …

### PR DESCRIPTION
**Pull request by be3jaay**
…in custom dialog.

### CODE REVIEW

File: apps/simulator/src/components/Dialog/CustomDialog.tsx [hunk 1]
```diff
@@ -15,6 +14,11 @@ interface CustomDialogProps {
   button?: ReactNode | ReactElement;
   ghostButton?: ReactNode | ReactElement;
   images?: ReactNode | ReactElement;
+  dialogSx?: object;
```
Comment (JezChristian) [inline apps/simulator/src/components/Dialog/CustomDialog.tsx:17]: Consider changing this, we don't really use object as the type as is,  try making it more defined

**Review by JezChristian — APPROVED** [review]

**Review by DropitupQwerty — APPROVED** [review]

File: apps/simulator/src/components/Dialog/CustomDialog.tsx [hunk 2]
```diff
@@ -13,12 +14,14 @@ export interface CustomDialogProps {
   icon?: ReactNode | ReactElement;
   button?: ReactNode | ReactElement;
   ghostButton?: ReactNode | ReactElement;
-  images?: ReactNode | ReactElement;
-  dialogSx?: object;
-  titleSx?: object;
-  contentSx?: object;
-  contentTextSx?: object;
-  actionsSx?: object;
+  sx?: {
```
Comment (JMSevilla) [inline apps/simulator/src/components/Dialog/CustomDialog.tsx:17]: This is miscommunication. This code shouldn't be an object please remove all the properties inside this object and permanently removed it. 

sx?: DialogProps['sx']

File: apps/simulator/src/components/Dialog/ReportIssue/ReportIssueDialog.tsx [hunk 3]
```diff
@@ -8,6 +8,7 @@ import { ReportSchema } from '@/core/schema/reportIssue/validation';
 import { CustomDialog } from '../CustomDialog';
 import { buttonStyle } from '../../Header/Header';
 import { ReportIssueForm } from './ReportIssueForm'
+
 export interface FormValues {
```
Comment (JMSevilla) [inline apps/simulator/src/components/Dialog/ReportIssue/ReportIssueDialog.tsx:12]: This is also bad practice. This is an interface a pure typescript but you are adding a JSX code inside of it.

File: apps/simulator/src/components/Dialog/ReportIssue/ReportIssueDialog.tsx [hunk 4]
```diff
@@ -40,7 +41,14 @@ export const ReportIssueDialog: React.FC = () => {
         open={open}
         aria-labelledby="responsive-dialog-title"
         title="Report an Issue"
-        titleSx={{ background: '#007AB7', color: '#F3F3F3', marginBottom: 2 }}
+        sx={{
```
Comment (JMSevilla) [inline apps/simulator/src/components/Dialog/ReportIssue/ReportIssueDialog.tsx:44]: We shouldn't add directly this styles inside an interface which is identified as pure typescript.

File: apps/simulator/src/components/Dialog/CustomDialog.tsx [hunk 5]
```diff
@@ -29,26 +32,20 @@ export const CustomDialog: React.FC<CustomDialogProps> = ({
...
-  titleSx,
-  contentSx,
-  contentTextSx,
-  actionsSx
+  sx = {}
 }) => {
   return (
-    <Dialog open={open} aria-labelledby="custom-dialog-title" sx={dialogSx}>
-      <DialogTitle id="custom-dialog-title" sx={titleSx}>
+    <Dialog open={open} aria-labelledby="custom-dialog-title" sx={sx.dialog}>
```
Comment (JMSevilla) [inline apps/simulator/src/components/Dialog/CustomDialog.tsx:38]: I suggest to not use sx if you want to create a multiple styles in your dialog. you can use the styled function from MUI system to make it more readable and easy to maintain. example : 

const CustomDialog = styled(Dialog)(({ theme }) => ({
  // your styles here
}));

const CustomDialogContent = styled(DialogContent)(({ theme }) => ({
  // your styles here
}));

**Review by JMSevilla — CHANGES_REQUESTED** [review]

File: apps/simulator/src/components/Dialog/CustomDialog.tsx [hunk 6]
```diff
@@ -3,27 +3,53 @@ import Dialog from '@mui/material/Dialog';
...
-    content?: SxProps<Theme>;
-    contentText?: SxProps<Theme>;
-    actions?: SxProps<Theme>;
-  };
-
+  className?: string;
+  sx?: DialogProps['sx'];
 }
 
+const DefaultDialog = styled(Dialog)(({ theme }) => ({
```
Comment (JMSevilla) [inline apps/simulator/src/components/Dialog/CustomDialog.tsx:22]: We don't want to add empty code as part of best practices if needed then we can just add it here.

File: apps/simulator/src/components/Dialog/CustomDialog.tsx [hunk 7]
```diff
@@ -3,27 +3,53 @@ import Dialog from '@mui/material/Dialog';
...
+    flexDirection: 'column',
+    background: '#B21E35',
+  },
+  '&.report-issue': {
+    background: '#007AB7',
+    marginBottom: 10
+  },
+}));
+
+const DefaultDialogContent = styled(DialogContent)(({ theme }) => ({
```
Comment (JMSevilla) [inline apps/simulator/src/components/Dialog/CustomDialog.tsx:41]: Same here

**Review by JMSevilla — CHANGES_REQUESTED** [review]

**Review by acostales08 — APPROVED** [review]

**Review by JMSevilla — APPROVED** [review]

**Pull request merged**